xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitlesLibass.cpp

```diff
@@ -54,20 +61,42 @@ CDVDSubtitlesLibass::CDVDSubtitlesLibass()
   if(!m_renderer)
     return;
 
-  //Setting default font to the Arial in \media\fonts (used if FontConfig fails)
-  const std::shared_ptr<CSettings> settings = CServiceBroker::GetSettingsComponent()->GetSettings();
-  strPath = URIUtils::AddFileToFolder("special://home/media/Fonts/", settings->GetString(CSettings::SETTING_SUBTITLES_FONT));
-  if (!XFILE::CFile::Exists(strPath))
-    strPath = URIUtils::AddFileToFolder("special://xbmc/media/Fonts/", settings->GetString(CSettings::SETTING_SUBTITLES_FONT));
-  int fc = !settings->GetBool(CSettings::SETTING_SUBTITLES_OVERRIDEASSFONTS);
-
   ass_set_margins(m_renderer, 0, 0, 0, 0);
   ass_set_use_margins(m_renderer, 0);
   ass_set_font_scale(m_renderer, 1);
 
+  int fontconfig = !settings->GetBool(CSettings::SETTING_SUBTITLES_OVERRIDEASSFONTS);
   // libass uses fontconfig (system lib) which is not wrapped
   //  so translate the path before calling into libass
-  ass_set_fonts(m_renderer, CSpecialProtocol::TranslatePath(strPath).c_str(), "Arial", fc, NULL, 1);
+  ass_set_fonts(m_renderer, CSpecialProtocol::TranslatePath(strPath).c_str(), "Arial", fontconfig,
+                nullptr, 1);
+}
+
+void CDVDSubtitlesLibass::AppendFont(std::string font) const
+{
+  std::string finalFontPath = URIUtils::AddFileToFolder("special://temp/fonts/", font);
+  if (!XFILE::CFile::Exists(finalFontPath))
```

Comment:
you can probably inverse the check here and return early to avoid the extra level of indentation